src/v/cluster/rm_stm.cc

```diff
@@ -913,30 +905,57 @@ ss::future<> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
         co_return;
     }
 
-    if (_mem_state.expected.contains(pid)) {
-        auto expiration_it = _mem_state.expiration.find(pid);
-        if (expiration_it != _mem_state.expiration.end()) {
-            if (expiration_it->second.deadline() > clock_type::now()) {
-                co_return;
-            }
+    auto expiration_it = _mem_state.expiration.find(pid);
+    if (expiration_it != _mem_state.expiration.end()) {
+        if (expiration_it->second.deadline() > clock_type::now()) {
+            co_return;
         }
     }
 
     _mem_state.expected.erase(pid);
 
-    auto batch = make_tx_control_batch(
-      pid, model::control_record_type::tx_abort);
-    auto reader = model::make_memory_record_batch_reader(std::move(batch));
-    auto r = co_await _c->replicate(
-      _insync_term,
-      std::move(reader),
-      raft::replicate_options(raft::consistency_level::quorum_ack));
-    if (!r) {
-        vlog(
-          clusterlog.warn,
-          "Error \"{}\" on aborting timed out pid: {}",
-          r.error(),
-          pid);
+    std::optional<prepare_marker> marker;
+    auto prepare_it = _mem_state.preparing.find(pid);
+    if (prepare_it != _mem_state.preparing.end()) {
+        marker = prepare_it->second;
+    }
+    prepare_it = _log_state.prepared.find(pid);
+    if (prepare_it != _log_state.prepared.end()) {
+        marker = prepare_it->second;
+    }
+
+    if (marker) {
+        auto r = co_await _tx_gateway_frontend.local().try_abort(
+          marker->tm_partition, marker->pid, marker->tx_seq, _sync_timeout);
+        if (r.ec == tx_errc::none) {
+            if (r.commited) {
+                auto batch = make_tx_control_batch(
+                  pid, model::control_record_type::tx_commit);
+                auto reader = model::make_memory_record_batch_reader(
+                  std::move(batch));
+                (void)co_await _c->replicate(
+                  _insync_term,
+                  std::move(reader),
+                  raft::replicate_options(raft::consistency_level::quorum_ack));
+            } else if (r.aborted) {
+                auto batch = make_tx_control_batch(
+                  pid, model::control_record_type::tx_abort);
+                auto reader = model::make_memory_record_batch_reader(
+                  std::move(batch));
+                (void)co_await _c->replicate(
+                  _insync_term,
+                  std::move(reader),
+                  raft::replicate_options(raft::consistency_level::quorum_ack));
+            }
+        }
+    } else {
+        auto batch = make_tx_control_batch(
+          pid, model::control_record_type::tx_abort);
+        auto reader = model::make_memory_record_batch_reader(std::move(batch));
+        (void)co_await _c->replicate(
```

Comment:
gate?

Reply:
oh... I see you are ignoring the return value not backgrounding the future. the idiomatic way to do this is `co_await c->replicate(...).discard_result()` which would be preferred here over the `(void)`